nemo-physical/src/datatypes/double.rs

```diff
@@ -20,30 +30,30 @@ impl Double {
     /// Wraps the given [f64]-`value` as a value over [Double].
     ///
     /// # Errors
-    /// The given `value` is [f64::NAN].
+    /// Returns an error if `value` is [f32::NAN] or infinite.
     pub fn new(value: f64) -> Result<Self, ReadingError> {
-        if value.is_nan() {
-            return Err(FloatIsNaN.into());
+        if !value.is_finite() {
+            return Err(ReadingError::InvalidFloat.into());
         }
 
         Ok(Self(value))
     }
 
-    /// Wraps the given [f64]-`value`, that is a number, as a value over [Double].
+    /// Wraps the given [f64]-`value` as a value over [Double].
     ///
     /// # Panics
-    /// The given `value` is [f64::NAN].
+    /// Panics if `value` is [f64::NAN] or not finite.
     pub fn from_number(value: f64) -> Self {
         if value.is_nan() {
-            panic!("The provided value is not a number (NaN)!")
+            panic!("floating point values must be finite")
         }
 
         Self(value)
     }
 
     /// Computes the absolute value.
     pub(crate) fn abs(self) -> Self {
-        Double::new(self.0.abs()).expect("Taking the absolute value cannot result in NaN")
+        Double::new(self.0.abs()).expect("operation returned in invalid float")
```

Comment:
at the very least, this should be grammatically correct, i.e., `operation returned an invalid float`.

Furthermore, we've generally been using [expect as precondition](https://doc.rust-lang.org/std/error/index.html#common-message-styles), so it should actually be something like `operation returns finite float`